optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,645 @@
+        )
+
+        ret = float(
+            TPESampler._compare(
+                samples=samples_below, log_l=log_likelihoods_below, log_g=log_likelihoods_above
+            )[0]
+        )
+        return math.exp(ret) if is_log else ret
+
+    def _sample_mo_categorical_index(
+        self,
+        study: "multi_objective.study.MultiObjectiveStudy",
+        trial: "multi_objective.trial.FrozenMultiObjectiveTrial",
+        distribution: distributions.CategoricalDistribution,
+        below: np.ndarray,
+        above: np.ndarray,
+    ) -> int:
+        choices = distribution.choices
+        below = list(map(int, below))
+        above = list(map(int, above))
+        upper = len(choices)
+        size = (self._n_ehvi_candidates,)
+
+        if self._weights is _default_weights_above:
+            weights_below = study._storage.get_trial(trial._trial_id).system_attrs[
+                _WEIGHTS_BELOW_KEY
+            ](len(below))
+        else:
+            weights_below = self._weights(len(below))
+        counts_below = np.bincount(below, minlength=upper, weights=weights_below)
+        weighted_below = counts_below + self._prior_weight
+        weighted_below /= weighted_below.sum()
+        samples_below = self._sample_from_categorical_dist(weighted_below, size)
+        log_likelihoods_below = TPESampler._categorical_log_pdf(samples_below, weighted_below)
+
+        weights_above = self._weights(len(above))
+        counts_above = np.bincount(above, minlength=upper, weights=weights_above)
+        weighted_above = counts_above + self._prior_weight
+        weighted_above /= weighted_above.sum()
+        log_likelihoods_above = TPESampler._categorical_log_pdf(samples_below, weighted_above)
+
+        return int(
+            TPESampler._compare(
+                samples=samples_below, log_l=log_likelihoods_below, log_g=log_likelihoods_above
+            )[0]
+        )
+
+    @staticmethod
+    def _compute_hypervolume(solution_set: np.ndarray, reference_point: np.ndarray) -> float:
+        return _hypervolume.WFG().compute(solution_set, reference_point)
+
+    def _solve_hssp(
+        self,
+        rank_i_loss_vals: np.ndarray,
+        rank_i_indices: np.ndarray,
+        subset_size: int,
+        reference_point: np.ndarray,
+    ) -> np.ndarray:
+        """Solve a hypervolume subset selection problem (HSSP) via a greedy algorithm.
+
+        This method is a 1-1/e approximation algorithm to solve HSSP.
+
+        For further information about algorithms to solve HSSP, please refer to the following
+        paper:
+
+        - `Greedy Hypervolume Subset Selection in Low Dimensions
+           <https://ieeexplore.ieee.org/document/7570501>`_
+        """
+        selected_vecs = []  # type: List[np.ndarray]
+        selected_indices = []  # type: List[int]
+        contributions = [
+            self._compute_hypervolume(np.asarray([v]), reference_point) for v in rank_i_loss_vals
+        ]
+        MARK_AS_SELECTED = -1
```

Comment:
According to the [PEP8](https://www.python.org/dev/peps/pep-0008/), the name of the variable in function should be lowercase. This variable is only used 2 times. How about removing it?